lib/key-agreement-key.js

```diff
@@ -23,25 +22,17 @@ exports.generateKeyPair = async ({keyId, type}) => {
   const now = Date.now();
   const meta = {created: now};
 
-  // only `X25519KeyAgreementKey2019` supported
-  if(type !== 'X25519KeyAgreementKey2019') {
-    throw new Error(`Unknown key type "${type}".`);
+  // only `X25519KeyAgreementKey2020` supported
+  if(type !== 'X25519KeyAgreementKey2020') {
+    throw new Error(`Unsupported key type "${type}".`);
   }
 
-  const {secretKey, publicKey} = nacl.box.keyPair();
-  const key = {
-    id: keyId,
-    type,
-    privateKeyBase58: base58.encode(Buffer.from(
-      secretKey.buffer, secretKey.byteOffset, secretKey.byteLength)),
-    publicKeyBase58: base58.encode(Buffer.from(
-      publicKey.buffer, publicKey.byteOffset, publicKey.byteLength))
-  };
+  const keyPair = await X25519KeyAgreementKey2020.generate({id: keyId});
 
   const record = {
     id: database.hash(keyId),
     meta,
-    key
+    key: keyPair.export({publicKey: true, privateKey: true})
```

Comment:
I'm going to assume the format here ends up being unchanged.

Reply:
Correct, yeah.